meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc

```diff
@@ -119,24 +145,39 @@ fakeroot python do_sign_class-target() {
     sb_sign(dir + grub_image, dir + grub_image, d)
     uks_sel_sign(dir + 'grub.cfg', d)
     uks_sel_sign(dir + 'boot-menu.inc', d)
+    uks_boot_sign(dir + 'grub.cfg', d)
+    uks_boot_sign(dir + 'boot-menu.inc', d)
 
     if d.getVar('UEFI_SB', True) == "1":
         uks_sel_sign(dir + 'efi-secure-boot.inc', d)
         uks_sel_sign(dir + 'password.inc', d)
+        uks_boot_sign(dir + 'efi-secure-boot.inc', d)
+        uks_boot_sign(dir + 'password.inc', d)
```

Comment:
Can we have one function we call with the filename to sign and it in turn does each type?  Thinking back to when I've had to extend / change this area for a different set of files to use, it's getting worse now if there's 2 calls for each file.

Reply:
I gave that some consideration earlier on, but the issue here is that these don't actually directly overlap.   For example, grub doesn't sign itself, where as there is a specific call to uks_sel_sign for the grub binary.  There is no two ways about it that it is ugly to have two completely separate signing mechanisms. 

My preference would be to get rid of both of them and use systemd-boot and add the signature functionality into systemd-boot, but that is a longer term project.

Reply:
I worry that we need to fix this now, not later.  Putting all of this into production before meant I added more files and such that needed signing.  And switching to systemd-boot could be interesting, but I think is still an aside from this problem.

Reply:
I really just want something like:
```
uks_secboot_sign(file, d)
    if d.getVar(... mok2verify ...):
        ...
    if d.getVar(... grub sigs ...):
        ...
```
So that every new file that needs signing calls one thing, not two things.